### ISSUE TITLE
Investigate Android sound code issues reported by Codacy

### ISSUE THREAD

**Issue by ann0see** [jamulussoftware/jamulus#2570]
**Describe the bug**

Codacy complains that the following variable is not used (and I don't see that this is false):

https://github.com/jamulussoftware/jamulus/blob/a9deb2259161b6cb86c12259f3e6f398189029ef/android/sound.cpp#L150

Also this warning: 
![image](https://user-images.githubusercontent.com/20726856/161150084-72b0f33c-caeb-4a8b-a2b3-067e8888a37a.png)

I assume this is a false positive?
**To Reproduce**

See https://app.codacy.com/gh/ann0see/jamulus/file/68862695471/issues/source?bid=21822857&fileBranchId=21822857

**Expected behavior**

No Codacy warnings

**Screenshots**

See above

**Operating system**

Android

**Version of Jamulus**

Latest

**Additional context**

@j-santander @sthenos could you please have a look at these warnings too?

**Comment by pljones** [comment]
If this is simple, we might as well try to resolve it in 3.11.0.

**Comment by pljones** [comment]
@ann0see I'm happy to pick this one up, unless you're planning to?

**Comment by ann0see** [comment]
Feel free to do so. It should be a quick fix.

**Comment by pljones** [comment]
`latencyMode` appears not to exist:
```
$ grep latencyMode android/*.* src/*.*
$
```

`iOboeBufferSizeMono` appears not to exist:
```
$ grep iOboeBufferSizeMono android/*.* src/*.*
$
```

Where on Github do I look to see Codacity output - is this on jamulussoftware/jamulus:main?

**Comment by ann0see** [comment]
> Where on Github do I look to see Codacity output - is this on jamulussoftware/jamulus:main?

No. It's just enabled for my repo for security reasons: https://app.codacy.com/gh/ann0see/jamulus/dashboard

**Comment by ann0see** [comment]
Codacy is an external proprietary tool while CodeQL is something supported by GitHub. Every plug-in in the official repo is another potential attack vector.

**Comment by pljones** [comment]
That doesn't answer my point.  If _anyone_ can use Codacity to expose flaws in Jamulus, should the Jamulus **team** not take on _responsibility_ by having the reports delivered to the team as part of each build?  Either there's a use for the tool, or there isn't, right?

**Comment by ann0see** [comment]
Yes. This is true. We can enable Codacy for the repo. We just didn't as it is a proprietary tool.

**Comment by mcfnord** [comment]
> [!NOTE]
> 📡 **STAND BY FOR AN LLM-AUTHORED MESSAGE.**

Revisiting this thread: one of the original Codacy findings was an unused-variable warning in the pre-Oboe `android/sound.cpp`. The tooling-policy question (Codacy vs. CodeQL) is separate and unresolved, but the underlying code defect was never fixed — and it carried over into the Oboe rewrite. It is still present on `main` today, in `src/sound/oboe/sound.cpp`:

```cpp
void CSound::warnIfNotLowLatency ( oboe::ManagedStream& stream, QString streamName )
{
    if ( stream->getPerformanceMode() != oboe::PerformanceMode::LowLatency )
    {
        QString latencyMode = ( stream->getPerformanceMode() == oboe::PerformanceMode::None ? "None" : "Power Saving" );
    }

    Q_UNUSED ( streamName );
}
```

This is more than a lint nit. The function is named `warnIfNotLowLatency` and is called on every stream open — for both `PlayStream` and `RecordStream` in `openStreams()`. It correctly detects the exact condition it exists for: the device or OEM silently downgrading the requested `LowLatency` performance mode, which raises round-trip audio latency for that user. It computes the human-readable mode string, and then discards it. There is no `qWarning`/`qInfo` call, so this real audio-quality condition has produced no log signal at all since the Oboe backend was introduced. `streamName` is explicitly `Q_UNUSED` in the same function, which confirms the no-op is total.

The fix is three lines — emit the warning the function already promises:

```cpp
void CSound::warnIfNotLowLatency ( oboe::ManagedStream& stream, QString streamName )
{
    if ( stream->getPerformanceMode() != oboe::PerformanceMode::LowLatency )
    {
        QString latencyMode = ( stream->getPerformanceMode() == oboe::PerformanceMode::None ? "None" : "Power Saving" );
        qWarning() << "Oboe " << streamName << " did not open in LowLatency performance mode (got: " << latencyMode << ")";
    }
}
```

This matches the existing `qInfo`/`qDebug` logging style already used in the same file (e.g. `printStreamDetails`), adds no dependency, and closes both the original Codacy warning and a genuine missing diagnostic.

I would be glad to open a small PR for this if you want it — flagging it here first, per the design-first preference, rather than opening one unprompted.

**Comment by pljones** [comment]
Where would an Android user see the information?

**Comment by mcfnord** [comment]
MY LLM WROTE:

@pljones — they wouldn't see it anywhere. `qWarning()` on Android goes to `logcat` and nowhere else, and `--log` is a `ServerOnlyOptions` entry (`main.cpp:361`) that makes the client `exit(1)` in client mode. So any diagnostic here is developer-only by construction. Your question answers itself, and it argues against what I proposed.

Worth knowing why it matters less than I thought: on real hardware the condition never occurs. Samsung Galaxy S22 Ultra, Android 16, Qt 5.15.2, CI APKs for both current `main` and a PR build — 6 `openStreams()` calls, 12 streams, **every one `PerformanceMode: LowLatency`, `SharingMode: Exclusive`**. The MMAP exclusive path, no fallbacks. One device, so not representative, but Android audio is working exactly as intended there.

So I'd **delete `warnIfNotLowLatency()`** rather than finish it. `printStreamDetails()` on the next line already logs `PerformanceMode` unconditionally, so nothing is lost, and it closes the Codacy finding with less code.

Unrelated to this issue but found alongside it: denying the microphone permission makes the app exit at `main.cpp:899` with no message at all. I'll file that separately with the trial data.

**Comment by ann0see** [comment]
> Unrelated to this issue but found alongside it: denying the microphone permission makes the app exit at main.cpp:899 with no message at all.

I think denied permissions on all OS leads to bad states.
